### DISCUSSION TITLE
Source IP overriden by NAT(VM IP where VIP Is attached)

### DISCUSSION THREAD

**Discussion by gabbar023** [istio/istio#49150] [Q&A]
We have created k8s cluster using 5 vms where we have used keepalive to create a floating VIP.
We are using this VIP as the istio LB IP.
My issue is All the requests coming to this Istio LB, we are not receiving the original source ip of the request. The source ip is getting overriden by the  VM IP of the server where this VIP is attached.
How can i get the original source ip and avoid this VM  IP overriden for istio gateway?
Can someone pls help?

**Comment by howardjohn** [comment]
https://tetrate.io/blog/istio-source-ip-transparency/ could help
